app/backend/app.py

```diff
@@ -205,15 +231,78 @@ def config():
             "showGPT4VOptions": current_app.config[CONFIG_GPT4V_DEPLOYED],
             "showSemanticRankerOption": current_app.config[CONFIG_SEMANTIC_RANKER_DEPLOYED],
             "showVectorOption": current_app.config[CONFIG_VECTOR_SEARCH_ENABLED],
+            "showUserUpload": current_app.config[CONFIG_USER_UPLOAD_ENABLED],
         }
     )
 
 
+@bp.post("/upload")
+@authenticated
+async def upload(auth_claims: dict[str, Any]):
+    request_files = await request.files
+    if "file" not in request_files:
+        # If no files were included in the request, return an error response
+        return jsonify({"message": "No file part in the request", "status": "failed"}), 400
+
+    user_oid = auth_claims["oid"]
+    file = request_files.getlist("file")[0]
+    user_blob_container_client: FileSystemClient = current_app.config[CONFIG_USER_BLOB_CONTAINER_CLIENT]
+    user_directory_client = user_blob_container_client.get_directory_client(user_oid)
+    try:
+        await user_directory_client.get_directory_properties()
+    except ResourceNotFoundError:
```

Comment:
I think we can do this in a follow up pr - but should we have some "rollback" code that cleans up partially uploaded files in case 1 of these steps fails?

Reply:
Just thinking the main creation steps that could fail:

* create_directory: If that fails, nothing's been uploaded, so they'd need to retry in the browser.
* upload_data: Ditto
* add_file: If that fails, the document is in their directory, but not in the index. That should be fine since the deletion code also deletes from file client before deleting from index, so they should be able to delete it still, and should be able to list it. But they should also get an error in the browser that'd prompt them to retry upload, and the next upload should override the previous.

I just tested out failing, and realized the UI wasn't showing a good error, so I'm amending that now.

Reply:
Failed upload:
![Screenshot 2024-04-09 at 2 34 53 PM](https://github.com/Azure-Samples/azure-search-openai-demo/assets/297042/4621224b-fa2b-43dc-be0a-87fe88816e35)

Successful upload after:
![Screenshot 2024-04-09 at 2 35 45 PM](https://github.com/Azure-Samples/azure-search-openai-demo/assets/297042/7d76bc96-cca9-491b-872b-5bfc41613d5d)